package.json

```diff
@@ -18,41 +18,44 @@
   },
   "private": true,
   "dependencies": {
-    "@nx/devkit": "17.2.8",
-    "@swc/helpers": "~0.5.2",
+    "@nx/devkit": "19.8.2",
+    "@swc/helpers": "0.5.13",
     "tslib": "^2.3.0"
   },
   "devDependencies": {
     "@commitlint/cli": "^18.6.0",
     "@commitlint/config-conventional": "^18.6.0",
+    "@eslint/eslintrc": "^2.1.1",
     "@jscutlery/semver": "^4.2.0",
-    "@nx/eslint": "17.2.8",
-    "@nx/eslint-plugin": "17.2.8",
-    "@nx/jest": "17.2.8",
-    "@nx/js": "17.2.8",
-    "@nx/plugin": "17.2.8",
-    "@nx/workspace": "17.2.8",
-    "@swc-node/register": "~1.6.7",
-    "@swc/cli": "~0.1.62",
-    "@swc/core": "~1.3.85",
-    "@types/jest": "^29.4.0",
-    "@types/node": "18.16.9",
-    "@typescript-eslint/eslint-plugin": "^6.9.1",
-    "@typescript-eslint/parser": "^6.9.1",
-    "eslint": "~8.48.0",
+    "@nx/eslint": "19.8.2",
+    "@nx/eslint-plugin": "19.8.2",
+    "@nx/jest": "19.8.2",
+    "@nx/js": "19.8.2",
+    "@nx/plugin": "19.8.2",
+    "@nx/workspace": "19.8.2",
+    "@swc-node/register": "1.9.2",
+    "@swc/cli": "0.3.14",
+    "@swc/core": "1.5.7",
+    "@types/jest": "29.5.13",
+    "@types/node": "18.19.9",
+    "@typescript-eslint/eslint-plugin": "7.18.0",
+    "@typescript-eslint/parser": "7.18.0",
```

Comment:
Now that you've added `typescript-eslint@^8.0.0`, I don't think you need these dependencies anymore.
This already includes these two dependencies in the corresponding version. Did Nx leave and update them?

Reply:
Nx didn't update them, I'll remove them as you suggest, I'll upgrade the remainder of the dependencies and upgrade to Nx 19.8.4 to maintain compatibility with 17-19.

Also fixing the `no-extra-semi` in the `eslint.config.js` as this is now in eslint stylistic

Reply:
A couple of eslint errors on both nx-go and nx-go-e2e, I will fix those:

/home/john/code/nx-go/packages/nx-go/eslint.config.js
  5:7  warning  'compat' is assigned a value but never used  @typescript-eslint/no-unused-vars

/home/john/code/nx-go/packages/nx-go/jest.config.ts
  1:1  warning  Unused eslint-disable directive (no problems were reported)

✖ 2 problems (0 errors, 2 warnings)
  0 errors and 1 warning potentially fixable with the `--fix` option.

Reply:
Also fixing verdaccio warning:
Determining test suites to run...(node:46509) [VERWAR002] VerdaccioWarning: The configuration property "logs" has been deprecated, please rename to "log" for future compatibility

Reply:
I've also fixed the pnpm cache in the pipeline.

I have a couple of remaining issues though, in the e2e test, I get this error:

Failed to start verdaccio: undefined
local registry exit 1

The tests are passing and the publish step is working, so not quite sure what's going on here.

Lastly the go cache in the pipeline has go caching turned on, but there is no go.mod, shall I just disable caching?

Reply:
I have quite a few problem with pnpm (lockfile v9) and the linter too. I'll also investigate on my side to find a solution and propose a PR to prepare this. It will allow to focus on the Nx 19 upgrade here 👍

Reply:
I don't think the pnpm version will matter too much if people are consuming the published package?  They'll use their own package manager at that point, so it probably only affects us?

I'll fix the go cache if you're happy with it being disabled in the pipeline to remove that warning.

That would just leave these errors:

Failed to start verdaccio: undefined
local registry exit 1

Still not sure why this is the case as it does run all tests successfully but it seems like it does a SIGTERM on the verdaccio process after all tests are complete and then the error occurs.

Reply:
Go cache is now set to false unless you want that changed, but it is currently unused, so this removes the warning about it in the pipeline.

Just this left!

![image](https://github.com/user-attachments/assets/6dacd555-41cd-4006-8903-6b45754dae38)

Reply:
The error is coming from here: https://github.com/nrwl/nx/blob/master/packages/js/src/executors/verdaccio/verdaccio.impl.ts